File: Online2/E/q.cpp
#include <bits/stdc++.h>

using namespace std;

int main()
{
    int n;
    cin >> n; n *= 2;
    
    int arr[n+1];
    for(int i=1; i<=n; i++) cin >> arr[i];
    sort(arr+1, arr+n+1);
    
    bool flag = 1;
    for(int i=1; i<=n; i++) if(arr[i]*arr[n-i+1] != arr[1]*arr[n]) flag = 0;
    
    if(flag)
    {
        cout << "Yes\n";
        
        for(int i=1; i<=n/2; i++) cout << arr[i] << ' ' << arr[n-i+1] << '\n';
    }
    else cout << "No\n";
}

File: Online2/E/q.py
n=int(input())
k=n
c=0
e=n*2-1
d=True
x=list(map(int,input().split()))
b=x.sort()
while d!=False and k!=0:
    if x[c]*x[e]==x[c+1]*x[e-1]:
        c=c+1
        e=e-1
    else:
        d=False
    k=k-1
if d==True:
    e=n*2-1
    print('Yes')
    for i in range(n):
        print(x[i],x[e])
        e=e-1
else:
    print('No')
